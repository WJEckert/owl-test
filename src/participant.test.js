import Participant from './participant.js';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import store from './STORE.js'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant session = {store.session} stage = {store.onStage} avatar = {store.avatar} name = {store.name}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  
it('renders the UI as expected', () => {
    const tree = renderer
    .create(<Participant inSession = {true} onStage = {true} avatar = 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1' name = 'Will'/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
    
