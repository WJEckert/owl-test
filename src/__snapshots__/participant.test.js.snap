// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="participant"
>
  <div
    className="image"
  >
    <img
      alt="Picture of person"
      className="participant-image"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="info"
  >
    <h2>
      Will
    </h2>
    <ul
      className="listOn"
    >
      <li>
         on stage
      </li>
    </ul>
  </div>
</div>
`;
